bittensor/_neuron/text/core_server/run.py

```diff
@@ -208,16 +202,28 @@ def synapse_check(synapse, hotkey):
         """
         Custom synapse function to blacklist certain synapse functions depending on the stake
         """
-        #TODO turn on synapse checking function
+        incoming_uid = metagraph.hotkeys.index(hotkey)
         if synapse.synapse_type == bittensor.proto.Synapse.SynapseType.TEXT_LAST_HIDDEN_STATE:
-            pass
+            
+            if metagraph.S[incoming_uid] < config.neuron.lasthidden_stake:
+                return False
             
         elif synapse.synapse_type == bittensor.proto.Synapse.SynapseType.TEXT_CAUSAL_LM:
-            pass
+
+            if metagraph.S[incoming_uid] < config.neuron.causallm_stake:
+                return False
             
+
+        elif synapse.synapse_type == bittensor.proto.Synapse.SynapseType.TEXT_CAUSAL_LM_NEXT:
+
+            if metagraph.S[incoming_uid] < config.neuron.causallmnext_stake:
+                return False
+
         elif synapse.synapse_type == bittensor.proto.Synapse.SynapseType.TEXT_SEQ_2_SEQ:
-            pass
-        
+
+            if (metagraph.S[incoming_uid] < config.neuron.seq2seq_stake) and (metagraph.S[incoming_uid,  uid]):
+                return False     
+
         return True
```

Comment:
should add an else return false here for the case where there is no matching synapse type

Reply:
Haha nice catch! Let me add that